src/main/java/com/drinchev/projectlabel/resources/ui/ProjectLabelBackgroundImage.java

```diff
@@ -0,0 +1,116 @@
+        setImageToIDE();
+    }
+
+    private boolean shouldShowBackgroundImage() {
+        return preferences.backgroundImagePosition() != BackgroundImagePosition.HIDDEN;
+    }
+
+    private void setImageToIDE() {
+        PropertiesComponent prop = projectLevelPropertiesComponent();
+        String opacity = String.valueOf(preferences.backgroundImageOpacity());
+        String imageProp = String.format("%s,%s,plain,%s", resultingImage, opacity, preferences.backgroundImagePosition().name().toLowerCase());
+        String prev_editor = prop.getValue(IdeBackgroundUtil.EDITOR_PROP);
+        prop.setValue(IdeBackgroundUtil.EDITOR_PROP, imageProp); // statt plain: tile (wiederholen), scale (zoomen)
+        String prev_frame = prop.getValue(IdeBackgroundUtil.FRAME_PROP);
+        prop.setValue(IdeBackgroundUtil.FRAME_PROP, imageProp);
+        LOG.warn("Previous editor image: " + prev_editor);
+        LOG.warn("Previous frame image: " + prev_frame);
+    }
+
+    private PropertiesComponent projectLevelPropertiesComponent() {
+        PropertiesComponent prop = project.getService(PropertiesComponent.class);
+        return prop;
+    }
+
+    public void updateImage() {
+        LOG.info("Updating project label background image.");
+        bufferedImage = null;
+        resultingImage = null;
+        showImage();
+    }
+
+    public void hideImage() {
+        PropertiesComponent prop = projectLevelPropertiesComponent();
+        if (!prop.getValue(IdeBackgroundUtil.EDITOR_PROP, "").contains("project-label") && !prop.getValue(IdeBackgroundUtil.FRAME_PROP, "").contains("project-label")) {
+            LOG.info("Not touching background image since not a project label image.");
+            return;
+        }
+        LOG.info("Hiding project label background image.");
+        prop.setValue(IdeBackgroundUtil.EDITOR_PROP, null);
+        prop.setValue(IdeBackgroundUtil.FRAME_PROP, null);
+    }
+
+
+    private void createImage() {
+        if (bufferedImage == null) {
+            try {
+                ProjectLabelAWTRenderer renderer = new ProjectLabelAWTRenderer(preferences);
+                Dimension preferredImageRation = renderer.getPreferredImageRatio();
+                final int paddingX = JBUIScale.scale(100);
+                final int paddingY = JBUIScale.scale(100);
+                // we want to have a 800x600 image, but prefer it to have the same ratio as the project label, so we scale it accordingly
+                final int targetWidth = 800;
+                final int targetHeight = (int) Math.round(targetWidth / preferredImageRation.getWidth() * preferredImageRation.getHeight());
+                bufferedImage = renderer.renderLabel(
+                        new Dimension(targetWidth+2*paddingX, targetHeight+2*paddingY),
+                        new Dimension(paddingX, paddingY));
+
+                Path filePath = Files.createTempFile("project-label", ".png");
+                filePath.toFile().deleteOnExit();
```

Comment:
Currently, I delete the file on JVM exit and regenerate a new one when opening the IDE again. Alternatively, we could store the image in a more persistent space and reuse it. Question there would be: where to store it? Or maybe the temp-file approach is good enough? 🤔

Reply:
Yeah I checked the docs and we are forced to use a file here. I guess it's okay to choose the temp-file approach :)